bee-api/bee-rest-api/src/endpoints/error.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use axum::{
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    Json,
+};
+use thiserror::Error;
+
+use crate::types::body::{DefaultErrorResponse, ErrorBody};
+
+// Errors that are returned to the user.
+#[derive(Error, Debug)]
+pub enum ApiError {
+    #[error("bad request: {0}")]
+    BadRequest(&'static str),
+    #[error("bad request: {0}")]
```

Comment:
Wrong comment

Reply:
why is it wrong?

Reply:
Idk it feels copy-pasted from the previous one, why not a more precise message ?

Reply:
The idea was that the part before `{0}` in the comments hints about the HTTP status code class. 
So to reflect:
- bad request
- not found
- service unavailable
- internal server error
- forbidden

And since `DependencyError` is a `bad request` i added that prefix. 
But what would could we add as a more precise message?